examples/azure/build-and-deploy-mender-artifact.yml

```diff
@@ -0,0 +1,110 @@
+      type: github
+      name: mendersoftware/mender-ci-workflows
+      ref: refs/heads/master
+      # GitHub service connection
+      # https://learn.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#github-service-connection
+      endpoint: github_service_connection
+
+
+#
+# Builds Mender Artifacts using 'mender-artifact' tool
+#
+jobs:
+- job: build_mender_artifact
+  pool:
+    vmImage: ubuntu-latest
+  # conrainer with pre-installed mender-artifact and mender-cli tools
+  container: mendersoftware/mender-ci-tools:mender-master
+  steps:
+  # Creates Mender Artifact, using randomly generated dummy name
+  # In this example, hello-mender.py represents a single file Python application that we have
+  # built and test in our CI and now we want to deploy into a device running Mender. We will
+  # package it as a Single File Application Update. For more information visit Mender docs:
+  # https://docs.mender.io/artifact-creation/create-an-artifact
+  - script: |
+      curl -O https://raw.githubusercontent.com/mendersoftware/mender-ci-workflows/master/examples/hello-mender.py
+      chmod 755 hello-mender.py
+      single-file-artifact-gen \
+        --device-type raspberrypi4 \
+        -o ${MENDER_ARTIFACT_NAME} \
+        -n ${MENDER_RELEASE_NAME} \
+        --software-name hello-mender \
+        --software-version 1.0 \
+        --dest-dir /usr/local/bin \
+        hello-mender.py
+    displayName: Build Mender Artifact
+  # Uploads Azure DevOps artifacts to Azure artifact storage to pass them to the further jobs
+  - task: PublishPipelineArtifact@1
+    inputs:
+      targetPath: $(MENDER_ARTIFACT_NAME)
+      artifactName: mender
+
+
+#
+# Uploads built Mender Artifacts to Mender server
+#
+- job: upload_mender_artifact
+  dependsOn:
+  - build_mender_artifact
+  pool:
+    vmImage: ubuntu-latest
+  steps:
+  # Downloads built in previous job Azure DevOps artifacts
+  - task: DownloadPipelineArtifact@2
+    inputs:
+      artifact: mender
+      patterns: $(MENDER_ARTIFACT_NAME)
+      path: $(System.DefaultWorkingDirectory)
+  # Uploads Mender Artifact to a Mender server
+  - template: templates/azure/mender-artifact-upload.yml@templates
+    parameters:
+      mender_uri: $(MENDER_SERVER_URL)
+      mender_pat: $(MENDER_SERVER_ACCESS_TOKEN)
+      mender_artifact: $(MENDER_ARTIFACT_NAME)
+
+
+#
+# Creates deployments on Mender server
+#
+- job: create_deployment
+  dependsOn:
+  - upload_mender_artifact
+  pool:
+    vmImage: ubuntu-latest
+  steps:
+  # Creates a deployment on a Mender server, using previously generated Mender deployment name
+  - template: templates/azure/mender-deployment-create.yml@templates
+    parameters:
+      mender_uri: $(MENDER_SERVER_URL)
+      mender_pat: $(MENDER_SERVER_ACCESS_TOKEN)
+      mender_deployment_name: $(MENDER_DEPLOYMENT_NAME)
+      mender_release_name: $(MENDER_RELEASE_NAME)
+      mender_deployment_group: $(MENDER_DEVICES_GROUP_NAME)
```

Comment:
Overall this example looks very nice! The template seems easy to use :medal_sports: